common/modules/gcp-project/audit.tf

```diff
@@ -11,19 +11,32 @@ resource "null_resource" "add_audit_config" {
 
   provisioner "local-exec" {
     command = <<EOF
-      echo "Applying audit config..."
+      EXIT_STATUS=0
 
-      auditConfigs=$(cat ${path.module}/resources/auditConfigs.json | jq -c -r .)
-      bindings=$(gcloud projects get-iam-policy ${google_project.project.project_id} --format json | jq -c -r .bindings)
+      auditConfigs=$(cat ${path.module}/resources/auditConfigs.json | jq -c -r -S .)
+      policy=$(gcloud projects get-iam-policy ${google_project.project.project_id} --format json)
+      policy_bindings=$(echo $policy | jq -c -r .bindings)
+      policy_auditConfigs=$(echo $policy | jq -c -r -S .auditConfigs)
 
-      jq -n -r \
-        --argjson auditConfigs "$auditConfigs" \
-        --argjson bindings "$bindings" \
-        '{"auditConfigs":$auditConfigs,"bindings":$bindings}' > ${path.module}/${google_project.project.project_id}.iam.policy.json
+      if [ "$(echo $auditConfigs | md5sum)" != "$(echo $policy_auditConfigs | md5sum)" ]; then
+        jq -n -r \
+          --argjson auditConfigs "$auditConfigs" \
+          --argjson bindings "$policy_bindings" \
+          '{"auditConfigs":$auditConfigs,"bindings":$bindings}' > ${path.module}/${google_project.project.project_id}.iam.policy.json
 
-      yes | gcloud projects set-iam-policy ${google_project.project.project_id} ${path.module}/${google_project.project.project_id}.iam.policy.json
+        echo "Applying audit configs..."
+        gcloud -q projects set-iam-policy ${google_project.project.project_id} ${path.module}/${google_project.project.project_id}.iam.policy.json
 
-      rm ${path.module}/${google_project.project.project_id}.iam.policy.json
+        if [ "$?" != "0" ]; then
+          EXIT_STATUS=1
```

Comment:
There must be a way to provide Terraform with the correct error code without explicitly returning it. What if another command, whose `$?` is not explicitly handed off, fails?

Can we `set -e`? Can we move this shell script to a separate file and run it with `sh -e script.sh`?

Reply:
Nice suggestion about `set -e`! Though it does not change much, because whatever fails on the way to the `gcloud projects set-iam-policy`, causes it to also fail. But it looks better anyway, so let me use that instead.